Temperature Monitor

griffinli opened the pull request:
## Describe your project in 1-3 sentences

A smart home temperature sensor that figures out how much I should open my window at night for the optimal sleep temperature.

## Hack Club Slack handle

@griffin

## Checklist before submitting proposal:

- [x] I am a current high school student (or younger).
- [x] I have submitted an RSVP at <https://hackclub.com/winter>
- [x] I am a member of the Hack Club Slack (Join at <https://hackclub.com/slack>)
- [x] I have self-reviewed my project proposal.

griffinli commented:
A proposal for a different project since #244 was declined

bellesea commented:
heyy @griffinli thanks so much for making a PR! we're back on tuesday and will start reviewing PRs in the order that they came on. 

to make the process smoother, please ensure you double checked that your file is properly formatted, you have all the parts you need, and that your plan is detailed enough for us to imagine how you're using the components in your budget to build your project :) 

and fyi, we're looking for projects where/that:
- you can spend at least 10 days building (from Feb 14th to 23rd)
- you are able to modify or make your own in some ways
- you can learn something new from
- involves working with electronics or software on a lower level than a pre-assembled kit of parts

and, we normally don’t approve projects where you buy consumer components w/o modification like keyboards (buying PCB, keycaps, switches, and assembling it), PCs (buying motherboard, cpu, memory and plugging it together), and kits (eg. closed-sourced consumer drone kits). 

_(note: from a first glance of your project i'm wondering: i love that it has such a cool twist but how do you know how much opening a window would affect temperatures in your room?)_

our Jan 15 11:59pm ET deadline is for PRs to come in and not for final approval so we'll work with you to either accept this project OR give you a chance to submit another PR. 

if you have any questions, please do not hesitate to ask her or DM me (i'm belle on slack), thank you!

cfanoulis commented on griffinli.md:
```diff
@@ -0,0 +1,33 @@
+| --------------- | ------------------------------------- | ------ |
+| Raspberry Pi 4  | https://www.amazon.com/Raspberry-Model-2019-Quad-Bluetooth/dp/B07TC2BK1X | $159.89  |
+| Raspberry Pi Case | https://www.amazon.com/Flirc-Raspberry-Pi-Case-Silver/dp/B07WG4DW52 | $15.95 |
+| Sandisk MicroSD Card | https://www.amazon.com/SanDisk-Extreme-microSDTM-Adapter-SDSQXCD-128G-GN6MA/dp/B09X7DNF6G | $21.00 |
```
Fairly certain raspberry pi's dont support 128gb cards :( As a raspberry user myself, I highly reccomend [this one](https://www.amazon.com/SanDisk-Extreme-microSDHC-Adapter-SDSQXWG-032G-ANCMA/dp/B01K3X3CXU) - I use it myself and its amazing

griffinli replied:
Looks like it's working on the pi 4 (https://forums.raspberrypi.com/viewtopic.php?t=268757) — I'll try it but if it doesn't work I can buy a smaller-sized one on my own

cfanoulis commented on griffinli.md:
```diff
@@ -0,0 +1,33 @@
+| Raspberry Pi Case | https://www.amazon.com/Flirc-Raspberry-Pi-Case-Silver/dp/B07WG4DW52 | $15.95 |
+| Sandisk MicroSD Card | https://www.amazon.com/SanDisk-Extreme-microSDTM-Adapter-SDSQXCD-128G-GN6MA/dp/B09X7DNF6G | $21.00 |
+| Raspberry Pi Power Supply | https://www.amazon.com/Raspberry-Model-Official-SC0218-Accessory/dp/B07W8XHMJZ | $7.95|
+| Temperature Sensor | https://www.adafruit.com/product/1782 | $4.95 |
```
I see you selected the adafruit breakout sensor - cool thing! however, it does require soldering - do you have the tools & consumables (solder, flux, wicks etc) necessary to solder?

griffinli replied:
My school has some that I can use!

cfanoulis replied:
great! mind including them in your budget list with a price of 0? thanks :)

cfanoulis commented on griffinli.md:
```diff
@@ -0,0 +1,33 @@
+| Breadboard | https://www.adafruit.com/product/64 | $4.95 |
+| Wires | https://www.adafruit.com/product/153 | $4.95 |
+| Adafruit Shipping | n/a | $6.10 |
+| Total           |                                       | $260.69 |
```
have you accounted for sales taxes in this total?

griffinli replied:
yeah — i'm in new hampshire so no tax

cfanoulis reviewed (changes requested):
hey griffinli, this is such a nice project idea :) please help me answer the following questions to get the grants goin'

griffinli reviewed (commented):
Reviewed changes

griffinli commented:
Looks like getting it to work in a smart home is much more straightforward with a separate Raspberry Pi so I added that to the proposal — I added the compute module so that I can also learn how to build with that. My friend found a site selling Raspberry Pi's for less than Amazon's price so I updated that as well, but I told him I wouldn't post it online and I'm also concerned with it selling out before I can get it for this project

cfanoulis commented:
hey - I'm not so certain a compute module is useful for this project

Do you want to try your hands on an arduino? I think it will be infinitely more easy to setup your project with that

griffinli commented:
hey — i think the smart home aspect does need a full computer (and i don't think an arduino would work). why wouldn't the compute module work?

cfanoulis reviewed (changes requested):
so after a bit more research i can see why you picked the raspi - didnt know homebridge didnt run on smol esps

however, I don't see a reason to include a compute module as well - as it practically serves the same purpose as the raspberry pi. I'd suggest you keep the raspi and remove the CM - it's more straight forward to work with (also small typical thing needed)

griffinli commented:
cool, thanks for taking a look

i feel like it would be interesting to try building with the compute module as well though (and it'll be my first time building with a normal raspberry pi so ideally i wouldn't want to only use one). can i get both so i can learn how to use them? i've done server and linux things on aws/etc so it shouldn't be too high of a learning curve.

griffinli commented:
the hackclub grant amount would also still be the same as before so i'm just spending some extra of my own funds to try also working with the compute module in this project

cfanoulis commented:
unfortunately, i cannot approve any extra hardware other than what's necessary for the project, and unfortunately it feels like 2 raspis are redundant :sob:

i understand wanting to build with a compute module, but i'd really like for you to have something awesome to ship on Feb 23rd - while having spent as little as your own money as we can (we're trying to lift you up here!)

griffinli commented:
ok, that makes sense. i think it would still be really interesting to build with a compute module though (and it also gives me the experience of building with a raspberry pi in general as well) so i've removed the standard pi from the proposal. does everything look good now?

cfanoulis reviewed (approved):
thanks for that - your grant is approved! I'd love to see you adding more tidbits in case you finish before Feb 23rd - maybe make your HVAC system turn on when its too hot? Or have the Pi send you a notification every morning with how hot your room was while you were sleeping

cfanoulis commented:
a @hackclub/bank-ops member will be in touch soon!

griffinli commented:
thank you! excited to build it, and i'll add some more features if i end up finishing early.

MelSmith104 reviewed (commented):
hey @griffinli! this project sounds awesome! im also very picky about the temperature of my room at night so i get where you're coming from. im dm'ing you on slack right now to chat about next steps!
